### PR TITLE
Documentation: Write secrets

### PR DESCRIPTION
**Pull request by ewjoachim**
### Check list:
- [x] Write documentation (we can help you doing so)

### CODE REVIEW

File: docs/howto/write.rst [hunk 1]
```diff
@@ -1,65 +1,121 @@
...
    Done
 
-   vault get third_secret
+   vault get http certificate
    ----BEGIN SECRET KEY----
    ...
 
-Identically, piping allows you to write the content of a file into the
-vault:
+Identically, stream redirection and pipes allows you to write the content of a file or
```
Comment (CorBott) [inline docs/howto/write.rst]: allow you?

File: docs/howto/write.rst [hunk 2]
```diff
@@ -1,65 +1,121 @@
...
    $ vault set mypath --prompt mykey
    Please enter a value for key `mykey` of `mypath`:
    Done
+
+Caveats
+-------
+
+Safe write
+~~~~~~~~~~
+Behaviour around overwriting secrets can be controlled, see :ref:`safe-write`.
```
Comment (CorBott) [inline docs/howto/write.rst]: Isn't this a ref to itself?

File: docs/howto/write.rst [hunk 3]
```diff
@@ -1,65 +1,121 @@
...
+Assuming we have a secret object at path ``a/b``:
+
+- It's forbidden to write a secret object at ``a``
+- It's forbidden to write a secret object at ``a/b/c``
+
+In other words, a same path cannot be a secret object and a ``folder``. This reproduces
+the classic file system paradigm, but not the classic URL paradigm where the notion of
+``folder`` does not exist.
+
+``vault-cli get-env --no-flat`` might have surprising results if this limitation was
```
Comment (CorBott) [inline docs/howto/write.rst]: was to be lifted?

File: docs/howto/write.rst (hunk 2 above)
Comment (ewjoachim) [reply docs/howto/write.rst]: No, it points to https://vault-cli.readthedocs.io/en/latest/howto/safe_write.html

**Comment by ewjoachim** [comment]
All done

File: docs/howto/write.rst [hunk 4]
```diff
@@ -1,65 +1,121 @@
...
 
-The special value ``--file=-`` means that the file is read from
+Identically, the special value ``--file=-`` means that the file is read from
 ``stdin``.
 
 
 Write a secret using an invisible input prompt
 ----------------------------------------------
 
 This will avoid your secrets to be displayed in plain text in your shell
```
Comment (CorBott) [inline docs/howto/write.rst]: This will avoid your secrets from being displayed

**Pull request merged**